### DISCUSSION TITLE
Recette v8.2.0

### DISCUSSION THREAD

**Discussion by alexisig** [MTES-MCT/sparte#638] [Recettes]
### Interface
* Refonte totale de l'interface des diagnostics en dashboard https://github.com/MTES-MCT/sparte/pull/559
    - Confirmer que tous les onglets fonctionnent pour un territoire AVEC OCS GE
        - Le contenu est lisible 
        - Les graphiques sont fonctionnels et exportables
        - Les boutons fonctionnent
        - Les documents exportables fonctionnent
        - Modifier les dates du diagnostic fonctionne
    - Confirmer que tous les onglets fonctionnent pour un territoire SANS OCS GE
        - Le contenu est lisible 
        - Les graphiques sont fonctionnels et exportables
        - Les boutons fonctionnent
        - Les documents exportables fonctionnent
        - Modifier les dates du diagnostic fonctionne
    - Confirmer que tous les onglets fonctionnent pour un territoire AVEC GPU
        - Le contenu est lisible 
        - Les graphiques sont fonctionnels et exportables
        - Les boutons fonctionnent
        - Les documents exportables fonctionnent
        - Modifier les dates du diagnostic fonctionne
    - Confirmer que tous les onglets fonctionnent pour un territoire SANS GPU
        - Le contenu est lisible 
        - Les graphiques sont fonctionnels et exportables
        - Les boutons fonctionnent
        - Les documents exportables fonctionnent
        - Modifier les dates du diagnostic fonctionne
    - Ajouter / Supprimer des territoires de comparaison et vérifier les graphiques de comparaison sur le site et sur les exports Word.
* Amélioration du système de recherche #583 
    - En cherchant la commune "Auch" le résultat apparait en premier
    - L'expérience de recherche est satisfaisante
### Données (rien à tester)
* Ajout des données OCS GE du département Corse-du-Sud (2A) https://github.com/MTES-MCT/sparte/pull/635
* Automatisation de l'ingestion des données de consommation https://github.com/MTES-MCT/sparte/pull/595
* Automatisation de l'ingestion des données des EPCI https://github.com/MTES-MCT/sparte/pull/601
* Amélioration des performances des models incremental de dbt https://github.com/MTES-MCT/sparte/pull/636
### Correctifs
* Autorisation des apostrophes dans les entrées utilisateurs https://github.com/MTES-MCT/sparte/pull/599
* Ajout d'une validation des mots de passe plus forte https://github.com/MTES-MCT/sparte/pull/610
* Lien vers FAQ mort https://github.com/MTES-MCT/sparte/issues/584
* Uniformiser les couleurs des graphiques https://github.com/MTES-MCT/sparte/issues/640
* Les graphiques de comparaison sur les exports word ne prennent pas en compte les modifications des territoires de comparaison https://github.com/MTES-MCT/sparte/issues/647
### Interne
* Simplification des notifications mattermost de crisp https://github.com/MTES-MCT/sparte/pull/631

**Tous les changements**: https://github.com/MTES-MCT/sparte/compare/V8.1.2...v8.2.0

**Comment by leabouche** [comment]
Onglet "Trajectoire ZAN"- Le bouton "Personnaliser mon objectif de réduction" ne fonctionne pas (testé sur plusieurs territoires)
<img width="540" alt="bouton personnaliser mon obj" src="https://github.com/user-attachments/assets/3b76fa8a-dc18-48f3-8e8c-01a5245d4a79">

**Reply by smdsgn** [reply]
corrigé #655

**Comment by leabouche** [comment]
La modification de la période d'analyse ne fonctionne pas. 
- Clic sur le bouton d'édition de la période d'analyse
- Je modifie l'année de début de 2011 à 2018 > enregistrer
- Modale "la période a été mise à jour"
--> Aucun effet sur la création d'un nouveau diag et la période affichée sur l'interface reste inchangée (période par défaut 2011-2022)

**Reply by leabouche** [reply]
actualiser la page met à jour les bonnes dates

**Reply by smdsgn** [reply]
Je n'ai pas le bug. À investiguer ensemble.

**Reply by JeremieSeguin** [reply]
C'est un gros 👍 que tout le diag n'ait pas besoin d'être rechargé ! 
Bravo @smdsgn

**Comment by leabouche** [comment]
Auch - Pas de données Zonages d'urbanisme dispo alors qu'on a les données GPU

**Reply by alexisig** [reply]
On a que des données pour le 33 en staging, on a prévu de limiter ce qui sort dans la recherche à ce qui est vraiment dispo pour que ce soit plus clair mais c'est pas encore en place

**Comment by leabouche** [comment]
Dans le menu: est-ce qu'on ferait pas apparaitre les onglets liés à l'OCSGE en grisé (ou autre) pour les territoires sans OCSGE, comme c'est le cas en prod?

**Reply by smdsgn** [reply]
Je ne suis pas trop fan en terme d'interface de griser des onglets. C'est pas hyper intuitif. Là au moins on peut mettre des messages plus explicites.

**Comment by leabouche** [comment]
Comparaison avec les territoires similaires - Bouton "Ajouter un nouveau territoire"
La recherche ne fonctionne pas, aucun résultat ne s'affiche

**Reply by alexisig** [reply]
corrigé #651

**Comment by leabouche** [comment]
Page Connexion - 2 Coquilles:
- Mot de pass**e** oublié ? 
- Réinitialisez-le

**Reply by alexisig** [reply]
corrigé #653

**Comment by leabouche** [comment]
Le connexion avec mes identifiants ne fonctionne pas

**Reply by alexisig** [reply]
La base de staging doit être issue d'un restore de la base de prod qui date d'avant ton arrivée, a priori c'est normal 👍

**Comment by leabouche** [comment]
Graphes Territoires similaires - Bugs dans l'affichage des courbes 

Concerne 2 graphes sur 3 (conso d'espace et conso d'espace proportionnelle à la surface)

- SCOTs : toutes les courbes s'affichent dans le même format (celui du SCOT recherché)
- SCOTs, Régions et Départements: Dans la légende, le territoire recherché apparaît deux fois 

![SCOT-1](https://github.com/user-attachments/assets/5fa3697f-d466-4961-9a1a-4a002dfb36e8)
![SCOT-2](https://github.com/user-attachments/assets/32d5a23b-f436-4298-9b2f-aad2fb1bdb80)
![DEPT-1](https://github.com/user-attachments/assets/21b627df-54a2-428c-b000-47b5b8654564)
![REGION](https://github.com/user-attachments/assets/b6bc2473-ea85-4179-bfed-f282ce13dcde)

**Reply by smdsgn** [reply]
corrigé #655

**Comment by JeremieSeguin** [comment]
Je demande si le contraste texte / fond est ok pour les mal voyants ou daltoniens ?

![image](https://github.com/user-attachments/assets/73b18843-5ed7-4ba2-982e-240709b404b4)

(Je n'en sais trop rien à vrai dire)

**Reply by leabouche** [reply]
oui je suis d'accord avec toi @JeremieSeguin - on peut peut être diminuer l'opacité du fond

**Reply by smdsgn** [reply]
Corrigé #692

**Comment by JeremieSeguin** [comment]
Liens "Qu'est-ce qu'un millésime" morts (404) 


**Pilotage territorial > Consommation d'espaces NAF** 
Tous les liens contenus dans les 5 encarts "Détails données et calcul" 

**Enjeux environnementaux > Artificialisation**
En début de section (sous le titre Etat des lieux du territoire au dernier millésime"

**Reply by smdsgn** [reply]
Corrigé #658

**Comment by JeremieSeguin** [comment]
Paramètres du diagnostic 

Les comportements des champs Nom* et Territoire semblent être inversés 

Le Nom* modifie le titre du diagnostic dans les fichiers téléchargés 
Le Champ Territoire modifie le titre du diagnostic sur la plateforme 

La plateforme indique l'inverse. 
Peut-être prévoir plus tard de n'avoir qu'un champ qui associe les deux comportements ?

**Reply by smdsgn** [reply]
En effet, c'est inutile d'avoir deux champs ici. On fera une passe sur les paramètres dans une US dédiée plus tard. Dans l'état ce n'est pas bloquant.

**Comment by alexisig** [comment]
En cliquant sur un zonage d'urbanisme dans l'explorateur : 

```js
bundle.prod.js:1 dsfr : version 1.12.1 
bundle.prod.js:1 dsfr : mode set to auto 
bundle.prod.js:1 dsfr : start 
bundle.prod.js:1 Uncaught ReferenceError: htmx is not defined
    at e.value (bundle.prod.js:1:1624165)
    at bundle.prod.js:1:1618211
    at Array.forEach (<anonymous>)
    at Bn.Map.<anonymous> (bundle.prod.js:1:1618179)
    at Bn.Map.r (bundle.prod.js:1:1566826)
    at Bn.Map.fire (bundle.prod.js:1:858680)
    at xr.click (bundle.prod.js:1:1480815)
    at HTMLDivElement.handleEvent (bundle.prod.js:1:1502437)
bundle.prod.js:1 Uncaught ReferenceError: htmx is not defined
    at e.value (bundle.prod.js:1:1624165)
    at bundle.prod.js:1:1618211
    at Array.forEach (<anonymous>)
    at Bn.Map.<anonymous> (bundle.prod.js:1:1618179)
    at Bn.Map.r (bundle.prod.js:1:1566826)
    at Bn.Map.fire (bundle.prod.js:1:858680)
    at xr.click (bundle.prod.js:1:1480815)
    at HTMLDivElement.handleEvent (bundle.prod.js:1:1502437)
bundle.prod.js:1 Uncaught ReferenceError: htmx is not defined
    at e.value (bundle.prod.js:1:1624165)
    at bundle.prod.js:1:1618211
    at Array.forEach (<anonymous>)
    at Bn.Map.<anonymous> (bundle.prod.js:1:1618179)
    at Bn.Map.r (bundle.prod.js:1:1566826)
    at Bn.Map.fire (bundle.prod.js:1:858680)
    at xr.click (bundle.prod.js:1:1480815)
    at HTMLDivElement.handleEvent (bundle.prod.js:1:1502437)
bundle.prod.js:1 Uncaught ReferenceError: htmx is not defined
    at e.value (bundle.prod.js:1:1624165)
    at bundle.prod.js:1:1618211
    at Array.forEach (<anonymous>)
    at Bn.Map.<anonymous> (bundle.prod.js:1:1618179)
    at Bn.Map.r (bundle.prod.js:1:1566826)
    at Bn.Map.fire (bundle.prod.js:1:858680)
    at xr.click (bundle.prod.js:1:1480815)
    at HTMLDivElement.handleEvent (bundle.prod.js:1:1502437)
value @ bundle.prod.js:1
(anonyme) @ bundle.prod.js:1
(anonyme) @ bundle.prod.js:1
r @ bundle.prod.js:1
fire @ bundle.prod.js:1
click @ bundle.prod.js:1
handleEvent @ bundle.prod.js:1


```

**Reply by smdsgn** [reply]
corrigé #668
